subprojects/core-api/src/main/java/org/gradle/api/artifacts/Configuration.java

```diff
@@ -272,7 +272,7 @@ public String determineName(Configuration c) {
      * @return The name of the associated upload task
      * @see org.gradle.api.tasks.Upload
      */
-    @Deprecated
+    @Deprecated // TODO:Finalize Upload Removal - Issue #21439
     String getUploadTaskName();
```

Comment:
Can we not remove this because of Spring too?

Reply:
I think so, it's used by the structuring-software-projects example (at least).  Running with `--stacktrace` doesn't help tell where, but that example is applying an older version of the Spring boot plugin.